birdhouse/components/weaver/config/magpie/weaver_hooks.py

```diff
@@ -50,15 +64,18 @@ def add_x_wps_output_context(request):
     # if explicitly provided, ensure it is permitted (admin allow any, otherwise self-user reference only)
     if header is not None:
```

Comment:
I feel like the logic of these `if` statements is unnecessarily complex. Can we simplify it to:

```py
if request.user is None:
    header = PUBLIC_USER_CONTEXT
elif header is None or (header != PUBLIC_USER_CONTEXT and not is_admin(request)):
    header = "users/" + str(request.user.id)
```

With the appropriate comments added in of course

Reply:
I agree for moving `if request.user is None:` to avoid repeting it, but the other `elif` is harder to read, and less diff when adding new combinations. Better to leave it sepated.